src/ado/lab_pipe.ado

```diff
@@ -0,0 +1,313 @@
+cap program drop   lab_pipe
+    program define lab_pipe
+
+    * Update the syntax. This is only a placeholder to make the command run
+    syntax , [ ///
+      PIPEValues(string) ignorepipes(string)  ///
+      outputlevel(string) truncate(string)   ///
+      ]
+
+    * Set defaults
+    if missing("`outputlevel'") local outputlevel "verbose"
+    if missing("`truncate'")    local truncate    "error"
+
+    if !(inlist("`outputlevel'","minimal","verbose","veryverbose")) {
+      noi di as error "{pstd}The value [`outputlevel'] in option {opt:outputlevel(`outputlevel')} is not a valid value. It may only be either minimal, verbose, or veryverbose.{p_end}"
+      error 198
+    }
+
+    if !(inlist("`truncate'","error","warning","prompt")) {
+      noi di as error "{pstd}The value [`truncate'] in option {opt:truncate(`truncate')} is not a valid value. It may only be either error, warning, or prompt.{p_end}"
+      error 198
+    }
+
+    * Initiate locals
+    local pipes_found     ""
+    local pipes_replaced  ""
+    local pipes_remaining ""
+    local truncate_all    "FALSE"
+
+    **************************************************
+    * Search for pipes
+    **************************************************
+
+    * Loop over all variables to list all pipes and all vars each are used for
+    foreach var of varlist _all {
+
+      * Reset locals used in serach
+      local is_pipe 0
+      local this_pipe ""
+
+      * Get this var's var labal and its length
+      local lab : variable label `var'
+      local lablen = strlen("`lab'")
+
+      * Parse the label
+      forvalues i = 1/`lablen' {
+        *Get first charachter
+        local c = substr("`lab'",`i',1)
+        * Test if beginning or end of pipe
+        if ("`c'"=="%") {
+          *Swithc is_pipe bool
+          local is_pipe = !`is_pipe'
+          *If no longer pipe, save this pipe and rest
+          if !(`is_pipe') {
+            * Test that pipe is a valid pipe - otherwise it is liekly not a pipe
+            local valid_pipe = regexm("`this_pipe'","^([a-z])([a-z0-9_]+)$")
```

Comment:
Consider using `ustrregexm` instead. While I've not found this documented anywhere, my experience is that this command does more than what's labelled on the tin: it also "upgrades" Stata's regex to a broader set than `regex*` allow. 

In general, this might be a bad move insofar as it limits users to Stata 14+ (?) when Unicode support was introduced. But for Survey Solutions users, this makes sense. The software [produces data in Stata 14 file format](https://docs.mysurvey.solutions/headquarters/export/stata-export-file-format/), so that those with older versions of Stata would have difficulties ingesting the files Survey Solutions exports.